src/AutoRest.CSharp/Common/Input/InputTypes/Serialization/TypeSpecInputModelTypeConverter.cs

```diff
@@ -87,7 +87,12 @@ public static InputModelType CreateModelType(ref Utf8JsonReader reader, string?
 
         private static InputModelType CreateInputModelTypeInstance(string? id, string? name, string? ns, string? accessibility, string? deprecated, string? description, string? usageString, string? discriminatorValue, string? discriminatorPropertyName, InputModelType? baseModel, List<InputModelProperty> properties, List<InputModelType> derivedModels, bool isNullable, ReferenceResolver resolver)
         {
-            name = name ?? throw new JsonException("Model must have name");
+            if (name == null)
+                throw new JsonException("Model must have name");
+
+            if (string.IsNullOrEmpty(name))
```

Comment:
```suggestion
            if (name == "")
```

null is checked above

Reply:
well I change it to name.Length == 0 since it is the best practice - actually for readability I would rather keep `IsNullOrEmpty` because it is basically doing the same check if you check the source of it.
https://stackoverflow.com/questions/7872633/most-performant-way-of-checking-empty-strings-in-c-sharp